lib/classes/Service.js

```diff
@@ -230,6 +232,22 @@ class Service {
     };
     warnOnDuplicateHandlers();
 
+    const provider = this.serverless.getProvider('aws');
+    if (provider) {
+      const stage = provider.getStage();
+      this.getAllFunctions().forEach(funcName => {
```

Comment:
Yes, it can be async technically.
But I think it is premature optimization and not necessary for this case.

Both `Service#getAllFunctions` and `Service#getAllEventsInFunction` used in validation are just iterating over functions (and its events).
I assume number of those are several dozens in most cases, and thousands in very extreme situation.
I believe Node.js is fast enough for such number of elements.

And, other validations in `Service#validate` are also iterating over functions synchronousely.
I can make them all asynchronously, but I think it is better to do so in another PR when users complains.

Reply:
@exoego 
OK 👍 Thank you for considering it 😄